SOOP/SOOP_CO2/incoming_handler.sh

```diff
@@ -83,49 +84,26 @@ handle_zip_file() {
     if [ $? -ne 0 ]; then
         rm -f $tmp_zip_manifest
         rm -rf --preserve-root $tmp_dir
-        file_error "Error unzipping"
+        file_error_and_report_to_uploader  $BACKUP_RECIPIENT "Error unzipping file '$file'"
     fi
 
     local nc_file
     nc_file=`grep ".*.nc" $tmp_zip_manifest | head -1`
     if [ $? -ne 0 ]; then
         rm -f $tmp_zip_manifest
         rm -rf --preserve-root $tmp_dir
-        file_error "Cannot find NetCDF file in zip bundle"
+        file_error_and_report_to_uploader $BACKUP_RECIPIENT "Cannot find NetCDF file in zip bundle '$file'"
     fi
 
     log_info "Processing '$nc_file'"
 
-    local tmp_nc_file=`make_writable_copy $tmp_dir/$nc_file`
-    local tmp_nc_file_with_sig
-    local checks='cf imos:1.3'
-
-    if is_imos_soop_co2_file $tmp_nc_file; then
-        tmp_nc_file_with_sig=`trigger_checkers_and_add_signature $tmp_nc_file $BACKUP_RECIPIENT $checks`
-    elif is_future_reef_map_file $tmp_nc_file; then
-        checks='cf'
-        tmp_nc_file_with_sig=`trigger_checkers_and_add_signature $tmp_nc_file $BACKUP_RECIPIENT $checks`
-    else
-        file_error "Not an underway CO2 file '$nc_file'"
-        rm -f $tmp_zip_manifest $tmp_nc_file_with_sig $tmp_nc_file
-        rm -rf --preserve-root $tmp_dir
-    fi
-    tmp_nc_file=$tmp_nc_file_with_sig
-
-    local path
-    path=`$SCRIPTPATH/dest_path.py $tmp_nc_file` || file_error "Cannot generate path for "`basename $tmp_nc_file`
-
-    s3_put $tmp_nc_file $path/`basename $nc_file` && rm -f $tmp_dir/$nc_file
+    local path_to_storage=`handle_netcdf_file $tmp_dir/$nc_file`
 
```

Comment:
Could be worth checking at this point that `$path_to_storage` is not an empty string?